crates/stackable-operator/src/role_utils.rs

```diff
@@ -144,6 +145,18 @@ pub struct CommonConfiguration<T> {
     #[serde(default)]
     #[schemars(schema_with = "raw_object_schema")]
     pub pod_overrides: PodTemplateSpec,
+
+    /// The minimum lifetime of secrets generated by the secret operator.
+    /// Some secrets, such as self signed certificates are constrained to a maximum lifetime by the
+    /// [SecretClass](DOCS_BASE_URL_PLACEHOLDER/secret-operator/secretclass) object it's self.
+    /// Currently this property covers self signed certificates but in the future it may be extended to other
+    /// secret types such as Kerberos keytabs.
+    #[serde(default = "default_min_secret_lifetime")]
+    pub min_secret_lifetime: Duration,
```

Comment:
In the corresponding [proposal](https://github.com/stackabletech/issues/issues/673) I try to explain the reasoning for naming. Other suggestions are welcome though.

As you mention, the idea is that Pod restarts due to secret expiration should be easily understood and transparent from a user's perspective regardless of the type of secrets.

Reply:
> As you mention, the idea is that Pod restarts due to secret expiration should be easily understood and transparent from a user's perspective regardless of the type of secrets.

Yeah that makes sense. On the other hand, I can see users wanting different lifetimes for different secrets (if there eventually will be multiple ones for a single group). So a less generic name might be something we have to think about.

Also, why do we prefix the field with `min_`? It is the lifetime of the secret, not a minimal lifetime which _could_ be longer. Pods will get restarted at _exactly_ at the lifetimes interval.

Reply:
Note: It might be worth to continue this discussion on the [proposal](https://github.com/stackabletech/issues/issues/673) directly. I'll move it to there.